compiler/codegen/OMRLinkage.hpp

```diff
@@ -132,6 +132,41 @@ class OMR_EXTENSIBLE Linkage
     */
    virtual void performPostBinaryEncoding() { }
 
+   /** @brief
+    *    Gets the offset (in number of bytes) from the stack frame pointer to the location on the stack where the first
+    *    (closest to the frame pointer) parameter is located. 
+    *
+    *  @details
+    *    For example given the following stack frame layout, and assuming a call to a function with 4 parameters:
+    *
+    *    @code
+    *        0x0588     +--------+
+    *                   |  ARG1  |
+    *        0x0580     +--------+
+    *                   |  ARG2  |
+    *        0x0578     +--------+
+    *                   |  ARG3  |
+    *        0x0570     +--------+   <- offset to first parm relative to the frame pointer (0x0570 - 0x0530 = 0x0040)
+    *                   |  ARG4  |
+    *        0x0568     +--------+ 
+    *                   |  ....  |
+    *        0x0530     +--------+   <- frame pointer
+    *                   |  ....  |
+    *        0x0500     +--------+   <- stack pointer
+    *    @endcode
+    *
+    *    The offset returned by this function (0x0040 in the above example) may not be the first argument (in argument
+    *    order) passed by the caller on the stack. It is up to the linkage to make use of this function to initialize
+    *    parameter offsets depending on the order in which the caller passes the arguments to the callee.
+    */
+   virtual int32_t getOffsetToFirstParm() const;
```

Comment:
This is the best I could come up with. I still don't like the name and think we should change it, but I can't think of a good name to call this API that resembles what it returns. Any help in naming this thing is appreciated!

Reply:
I've been thinking on this for a while and I haven't been able to come up with a particularly good name either considering that this could refer to the offset of the first or last parameter in the function's signature that was passed via the stack depending on whether the `RightToLeft` flag is set or not... Nothing I've been able to come up with seems to be any better than what's already here...